homeassistant/components/fritz/config_flow.py

```diff
@@ -244,3 +267,54 @@ async def async_step_import(self, import_config):
                 CONF_PORT: import_config.get(CONF_PORT, DEFAULT_PORT),
             }
         )
+
+
+class FritzBoxToolsOptionsHandler(OptionsFlow):
+    """Handle a option flow."""
+
+    def __init__(self, config_entry: ConfigEntry) -> None:
+        """Initialize options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle options flow."""
+        fritz_tools: FritzBoxTools = self.hass.data[DOMAIN][self.config_entry.entry_id][
+            FRITZ_TOOLS
+        ]
+        devices: dict[str, str] = {
+            mac: fritz_device_name(device)
+            for mac, device in fritz_tools.devices.items()
+        }
+
+        if user_input is not None:
+            if user_input[CONF_ADD_ALL_TRACKER]:
+                user_input[CONF_ADD_NEW_TRACKER] = True
+                user_input[CONF_SELECTED_DEVICES] = list(devices.keys())
+                pass
+            return self.async_create_entry(title="", data=user_input)
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_ADD_ALL_TRACKER,
+                    default=self.config_entry.options.get(
+                        CONF_ADD_ALL_TRACKER, DEFAULT_ADD_ALL_TRACKER
+                    ),
+                ): bool,
+                vol.Required(
+                    CONF_ADD_NEW_TRACKER,
+                    default=self.config_entry.options.get(
+                        CONF_ADD_NEW_TRACKER, DEFAULT_ADD_NEW_TRACKER
+                    ),
+                ): bool,
+                vol.Required(
+                    CONF_SELECTED_DEVICES,
```

Comment:
This PR follows this approach -> https://developers.home-assistant.io/docs/entity_registry_disabled_by/#integrations-offering-options-to-control-disabled_by

Reply:
That is not meant to be done per device or entity, but should be some kind of group. We can already disable entities per entity.

Reply:
The intention is, that not necessarily all known devices should be added to device registry, if only one or two will be used.
In my case, my Fritzbox knows about 40 devices (_tv, radio, alexa, laptops, pc, FireTV, tablet, NAS, some VMs, guest clients, and my smartphone_) but I only want to track my smartphone.

Reply:
Yes, the entity won't be tracked if it's disabled. This is why we added that option in the first place. So that we wouldn't have to add this option specifically to every integration.

Reply:
Mhhh ... so if `entity_registry_enabled_default` will be `False` for all `device_tracker` entities, will related devices (_regarding to `device_info`_) still be added to device registry with disabled device_tracker entities, or does neither the devices as also the entities were registered?

Reply:
Devices can be disabled too. If a device is disabled, all its entities are also disabled. So the user can disable a device and all its entities or go down a level and disable specific entities.

The devices and entities will still be registered in the device and entity registries. But that doesn't impact performance really. So there's no point in avoiding that.

Reply:
> The devices and entities will still be registered in the device and entity registries. But that doesn't impact performance really. So there's no point in avoiding that.

ahhh ... ok, that is good to know, thanks :+1:

Reply:
@chemelli74 What do you prefer
1. add disabled devices, which inherits to connected device_tracker entity
2. add enabled device, but with disabled device_tracker entity

Reply:
I would vote for 2. as well

Reply:
The agreed solution will be implemented with #50791.
This PR will be closed unmerged.